### PR TITLE
Kafka readers and writers

### PR DESCRIPTION
**Pull request by sglvladi**
This PR contributes the following classes:
- `KafkaDetectionReader`: A detection reader that reads detections from a Kafka broker
- `KafkaGroundTruthReader`: A ground truth reader that reads ground truths from a Kafka broker
- `KafkaWriter`: A simple Kafka writer that writes data to a Kafka topic.

### CODE REVIEW

**Review by jswright-dstl — APPROVED** [review]

File: setup.cfg [hunk 1]
```diff
@@ -27,6 +27,7 @@ install_requires =
     rtree
     scipy
     utm
+    confluent-kafka
```
Comment (sdhiscocks) [inline setup.cfg:30]: I think this should be optional extra.

File: stonesoup/reader/kafka.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+from datetime import datetime, timedelta
+from math import modf
+from queue import Empty, Queue
+from threading import Thread
+from typing import Dict, List, Collection
+
+import numpy as np
+from confluent_kafka import Consumer
+from dateutil.parser import parse
+from stonesoup.base import Property
```
Comment (sdhiscocks) [inline stonesoup/reader/kafka.py:12]:
```suggestion
from dateutil.parser import parse

from stonesoup.base import Property
```

File: stonesoup/writer/kafka.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import json
+from typing import Dict
+
+from confluent_kafka import Producer
+from stonesoup.base import Property
```
Comment (sdhiscocks) [inline stonesoup/writer/kafka.py:5]:
```suggestion
from confluent_kafka import Producer

from stonesoup.base import Property
```

**Review by sdhiscocks — CHANGES_REQUESTED** [review]
Looks good, but I think `confluent_kafka` should be optional.

Future aspiration would be to use JSON Path syntax to allow selection of fields, which would enable this to be used more easily with GeoJSON for example.

**Review by jswright-dstl — APPROVED** [review]

**Pull request merged**